Source/WebKit/UIProcess/API/ios/WKWebViewIOS.mm

```diff
@@ -2530,6 +2535,27 @@ - (BOOL)_shouldDeferGeometryUpdates
     return _perProcessState.liveResizeParameters || _perProcessState.dynamicViewportUpdateMode != WebKit::DynamicViewportUpdateMode::NotResizing;
 }
 
+- (std::optional<WebKit::VisibleContentRectUpdateInfo>)_createVisibleContentRectUpdateInfo
+{
+    auto viewStability = _viewStabilityWhenVisibleContentRectUpdateScheduled;
+
+    CGRect visibleRectInContentCoordinates = [self _visibleContentRect];
+
+    UIEdgeInsets computedContentInsetUnadjustedForKeyboard = [self _computedObscuredInset];
+    if (!_haveSetObscuredInsets)
+        computedContentInsetUnadjustedForKeyboard.bottom -= _totalScrollViewBottomInsetAdjustmentForKeyboard;
+
+    CGFloat scaleFactor = contentZoomScale(self);
+    CGRect unobscuredRect = UIEdgeInsetsInsetRect(self.bounds, computedContentInsetUnadjustedForKeyboard);
+    WebCore::FloatRect unobscuredRectInContentCoordinates = WebCore::FloatRect(_perProcessState.frozenUnobscuredContentRect ? _perProcessState.frozenUnobscuredContentRect.value() : [self convertRect:unobscuredRect toView:_contentView.get()]);
+    if (![_contentView sizeChangedSinceLastVisibleContentRectUpdate])
+        unobscuredRectInContentCoordinates.intersect([self _contentBoundsExtendedForRubberbandingWithScale:scaleFactor]);
+
+    auto contentInsets = [self currentlyVisibleContentInsetsWithScale:scaleFactor obscuredInsets:computedContentInsetUnadjustedForKeyboard];
+
+    return [_contentView createVisibleContentRectUpdateInfoFromVisibleRect:visibleRectInContentCoordinates unobscuredRect:unobscuredRectInContentCoordinates contentInsets:contentInsets unobscuredRectInScrollViewCoordinates:unobscuredRect obscuredInsets:_obscuredInsets unobscuredSafeAreaInsets:[self _computedUnobscuredSafeAreaInset] inputViewBounds:_inputViewBoundsInWindow scale:scaleFactor minimumScale:[_scrollView minimumZoomScale] viewStability:viewStability enclosedInScrollableAncestorView:scrollViewCanScroll([self _scroller]) sendEvenIfUnchanged:_alwaysSendNextVisibleContentRectUpdate];
```

Comment:
Please put each argument on its own line for our future sanity when we edit this.

Reply:
I had that originally but then the style checker got mad :( will change it back